cabal-lint/README.md

```diff
@@ -0,0 +1,19 @@
+The IOG Consensus team uses this `cabal-lint` tool to ensures some invariants among its `.cabal` files.
+
```

Comment:
Question for the reviewers: what other invariants do we want to enforce as part of this PR?

Reply:
Done 👍